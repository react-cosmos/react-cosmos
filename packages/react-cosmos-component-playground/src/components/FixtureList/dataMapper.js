import set from 'lodash.set';

const dataObjectToNestedArray = (base, path = '') => {
  const returnChildren = [];
  for (const key in base) {
    if (typeof base[key] === 'object') {
      const newPath = path ? `${path}/${key}` : key;
      const children = dataObjectToNestedArray(base[key], newPath);
      returnChildren.push({
        name: key,
        expanded: true,
        children
        // urlParams: { component: newPath }
      });
    } else {
      const fixtures = base.map(fixture => ({
        name: fixture,
        urlParams: { component: path, fixture }
      }));
      return fixtures;
    }
  }
  return returnChildren;
};

const fixturesToTreeData = fixtures => {
  const components = Object.keys(fixtures);
  const data = {};

  components.forEach(componentPath => {
    const pathArray = componentPath.split('/');
    const fixturesAtPath = fixtures[componentPath];
    set(data, pathArray, fixturesAtPath);
  });

  return dataObjectToNestedArray(data);
};

export default fixturesToTreeData;

// const input = {
//   "dirA/Component1": ["fixtureA", "fixtureB"],
//   "dirB/Component2": ["fixtureA", "fixtureB"],
//   "dirB/Component3": ["fixtureA", "fixtureB"],
//   "dirB/subdirA/Component4": ["fixtureA", "fixtureB"],
// };

// const output = [
//   {
//     name: 'dirA',
//     expanded: true,
//     children: [
//       {
//         name: 'Component1',
//         expanded: true,
//         children: [
//           {
//             name: 'fixtureA',
//             urlParams: {
//               component: 'dirA/Component1',
//               fixture: 'fixtureA'
//             }
//           },
//           {
//             name: 'fixtureB',
//             urlParams: {
//               component: 'dirA/Component1',
//               fixture: 'fixtureB'
//             }
//           }
//         ]
//       }
//     ]
//   },
//   {
//     name: 'dirB',
//     expanded: true,
//     children: [
//       {
//         name: 'Component2',
//         expanded: true,
//         children: [
//           {
//             name: 'fixtureA',
//             urlParams: {
//               component: 'dirB/Component2',
//               fixture: 'fixtureA'
//             }
//           },
//           {
//             name: 'fixtureB',
//             urlParams: {
//               component: 'dirB/Component2',
//               fixture: 'fixtureB'
//             }
//           }
//         ]
//       },
//       {
//         name: 'Component3',
//         expanded: true,
//         children: [
//           {
//             name: 'fixtureA',
//             urlParams: {
//               component: 'dirB/Component3',
//               fixture: 'fixtureA'
//             }
//           },
//           {
//             name: 'fixtureB',
//             urlParams: {
//               component: 'dirB/Component3',
//               fixture: 'fixtureB'
//             }
//           }
//         ]
//       },
//       {
//         name: 'subdirA',
//         expanded: true,
//         children: [
//           {
//             name: 'Component4',
//             expanded: true,
//             children: [
//               {
//                 name: 'fixtureA',
//                 urlParams: {
//                   component: 'dirB/subdirA/Component4',
//                   fixture: 'fixtureA'
//                 }
//               },
//               {
//                 name: 'fixtureB',
//                 urlParams: {
//                   component: 'dirB/subdirA/Component4',
//                   fixture: 'fixtureB'
//                 }
//               }
//             ]
//           }
//         ]
//       }
//     ]
//   }
// ];
