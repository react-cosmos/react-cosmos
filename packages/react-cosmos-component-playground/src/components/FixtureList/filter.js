// Adapted from https://github.com/alexcurtis/react-treebeard
import { match } from 'fuzzaldrin-plus';

// Helper functions for search filtering
function matcher(filterText, node) {
  const matchText = node.urlParams
    ? `${node.urlParams.component}${node.urlParams.fixture}`
    : node.name;
  return match(matchText, filterText).length > 0;
}

function findNode(node, filterText) {
  return (
    matcher(filterText, node) || // i match
    (node.children && // or i have decendents and one of them match
      node.children.length &&
      Boolean(node.children.find(childNode => findNode(childNode, filterText))))
  );
}

function filterNode(node, filterText) {
  // If im an exact match then all my children get to stay
  if (matcher(filterText, node) || !node.children) {
    return { ...node, expanded: true };
  }
  // If not then only keep the ones that match or have matching descendants
  const filteredChildren = filterNodeArray(node.children, filterText);
  return { ...node, expanded: true, children: filteredChildren };
}

export default function filterNodeArray(nodeList, filterText) {
  return nodeList
    .filter(node => findNode(node, filterText))
    .map(node => filterNode(node, filterText));
}
