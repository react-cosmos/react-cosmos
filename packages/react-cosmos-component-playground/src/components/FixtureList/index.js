import React, { Component } from 'react';
import { objectOf, arrayOf, shape, string, func, bool } from 'prop-types';
import { match } from 'fuzzaldrin-plus';
import { uri } from 'react-querystring-router';
import { SearchIcon } from '../SvgIcon';
import TreeView from './components/Tree';
import styles from './index.less';

const KEY_S = 83;
const KEY_ESC = 27;
export default class FixtureList extends Component {
  state = {
    searchText: '',
    selectedPath: '',
    filteredFixtures: {}
  };

  componentWillMount() {
    this.updateTreePropsWithText();
  }

  componentDidMount() {
    window.addEventListener('keydown', this.onWindowKey);
  }

  componentWillReceiveProps(nextProps) {
    const { nextFixtures, nextUrlParams } = nextProps;
    const { fixtures, urlParams } = this.props;
    if (nextFixtures !== fixtures || nextUrlParams !== urlParams) {
      this.updateTreePropsWithText();
    }
  }

  componentWillUnmount() {
    window.removeEventListener('keydown', this.onWindowKey);
  }

  onWindowKey = e => {
    const isFocused = this.searchInput === document.activeElement;

    if (e.keyCode === KEY_S && !isFocused) {
      // Prevent entering `s` in the search field along with focusing
      e.preventDefault();

      this.searchInput.focus();
    } else if (e.keyCode === KEY_ESC && isFocused) {
      const searchText = '';
      this.updateTreePropsWithText(searchText);

      this.searchInput.blur();
    }
  };

  onChange = e => {
    const searchText = e.target.value;
    this.updateTreePropsWithText(searchText);
  };

  onFixtureClick = ({ value }) => {
    const { component, fixture } = value;
    const href = uri.stringifyParams({
      ...this.props.urlParams,
      component,
      fixture
    });
    this.props.onUrlChange(href);
  };

  updateTreePropsWithText = (searchText = this.state.searchText) => {
    const { fixtures, urlParams } = this.props;
    console.log(urlParams, getSelectedPath(urlParams));
    this.setState(() => ({
      searchText,
      selectedPath: getSelectedPath(urlParams),
      filteredFixtures: getFilteredFixtures(fixtures, searchText)
    }));
  };

  render() {
    const { selectedPath, filteredFixtures, searchText } = this.state;
    return (
      <div className={styles.root}>
        <div className={styles.searchInputContainer}>
          <input
            className={styles.searchInput}
            placeholder="Search..."
            value={searchText}
            onChange={this.onChange}
            ref={node => {
              this.searchInput = node;
            }}
          />
          <SearchIcon />
        </div>
        <TreeView
          fixtures={filteredFixtures}
          selectedPath={selectedPath}
          onLeafClick={this.onFixtureClick}
        />
      </div>
    );
  }
}

FixtureList.propTypes = {
  fixtures: objectOf(arrayOf(string)).isRequired,
  urlParams: shape({
    component: string,
    fixture: string,
    editor: bool,
    fullScreen: bool
  }).isRequired,
  onUrlChange: func.isRequired
};

export function getSelectedPath(urlParams) {
  const { component, fixture } = urlParams;
  return component && fixture ? `${component}/${fixture}` : undefined;
}

export function getFilteredFixtures(fixtures, searchText) {
  if (searchText.length < 2) {
    return fixtures;
  }

  const components = Object.keys(fixtures);

  return components.reduce((acc, componentName) => {
    const matchingFixtures = fixtures[componentName].filter(
      fixtureName =>
        match(`${componentName}${fixtureName}`, searchText).length !== 0
    );

    // Do not render the component if it has no matching fixtures
    if (matchingFixtures.length === 0) {
      return acc;
    }

    acc[componentName] = matchingFixtures;

    return acc;
  }, {});
}
