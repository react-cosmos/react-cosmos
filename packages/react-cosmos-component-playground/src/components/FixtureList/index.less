@import '../../utils/common.less';

@fixture-line-height: 30px;
@fixture-component-font-size: 16px;

.root {
  display: flex;
  flex-direction: column;
  flex: 1;
  font-family: @default-font-family;
  -webkit-font-smoothing: antialiased;
  overflow-y: hidden;
}

.searchInputContainer {
  flex-shrink: 0;
  position: relative;
  border-bottom: 1px solid @left-nav-bgcolor-selected;

  svg {
    .vertical-center();

    left: @default-spacing;
    width: 22px;
    height: 22px;
    fill: @left-nav-color;
  }
}

.searchInput {
  display: block;
  height: @fixture-line-height;
  width: 100%;
  padding: 0 @default-spacing;
  padding-left: @fixture-line-height + @default-spacing;
  background-color: @left-nav-bgcolor;
  color: @left-nav-color;
  border: 0;
  box-shadow: none;
  font-size: @editor-font-size;
  transition: background-color .3s ease;

  &:focus,
  &:active {
    outline: none;
    background-color: @left-nav-bgcolor-selected;
  }

  &::placeholder {
    color: fade(@left-nav-color-selected, 30%);
  }
}

.list {
  background: @default-darken-color;
  overflow-x: hidden;
  overflow-y: auto;
  flex: 1;
}

.component {
}

.componentName {
  display: flex;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-items: center;
  height: @fixture-line-height + @default-spacing;
  margin: 0;
  padding: @default-spacing @default-spacing 0 @default-spacing;
  color: @left-nav-color-selected;
  font-size: @fixture-component-font-size;
  line-height: @fixture-line-height;
  cursor: default;

  svg {
    flex-shrink: 0;
    width: 20px;
    height: 20px;
    fill: @left-nav-color-selected;
    margin-right: 0.5 * @default-spacing;
  }

  span {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}

.fixture {
  display: block;
  height: @fixture-line-height;
  padding: 0 @default-spacing;
  color: @left-nav-color;
  line-height: @fixture-line-height;
  text-decoration: none;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &:hover {
    background: @left-nav-bgcolor-selected;
  }
}

.fixtureSelected {
  background: @left-nav-bgcolor-selected;
  color: @left-nav-color-selected;
}
