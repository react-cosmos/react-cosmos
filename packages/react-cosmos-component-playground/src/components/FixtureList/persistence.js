import localForage from 'localforage';
import flatMapDeep from 'lodash.flatmapdeep';

const TREE_EXPANSION_STATE = '__cosmos__tree-expansion-state';

export function getSavedExpansionState(projectKey) {
  return localForage.getItem(`${TREE_EXPANSION_STATE}-${projectKey}`);
}

function setSavedExpansionState(projectKey, value) {
  return localForage.setItem(`${TREE_EXPANSION_STATE}-${projectKey}`, value);
}

export function updateLocalToggleState(projectKey, path, expanded) {
  getSavedExpansionState(projectKey).then(currentState => {
    setSavedExpansionState(projectKey, {
      ...currentState,
      [path]: expanded
    });
  });
}

function fixtureTreeToPathArray(fixtureTree) {
  return flatMapDeep(fixtureTree, node => {
    const result = [];
    if (node.path) {
      result.push(node.path);
    }
    if (node.children) {
      result.push(fixtureTreeToPathArray(node.children));
    }
    return result;
  });
}

export function pruneUnusedExpansionState(
  projectKey,
  savedExpansionState,
  fixtureTree
) {
  const pathArray = fixtureTreeToPathArray(fixtureTree);
  const prunedExpansionState = { ...savedExpansionState };
  const paths = Object.keys(savedExpansionState);
  paths.forEach(path => {
    if (pathArray.indexOf(path) === -1) {
      delete prunedExpansionState[path];
    }
  });
  setSavedExpansionState(projectKey, prunedExpansionState);
}
