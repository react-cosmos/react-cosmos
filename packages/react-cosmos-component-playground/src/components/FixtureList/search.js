// Adapted from https://github.com/alexcurtis/react-treebeard
import { match } from 'fuzzaldrin-plus';

// Helper functions for search filtering
const matcher = (filterText, node) => {
  const matchText = node.urlParams
    ? `${node.urlParams.component}${node.urlParams.fixture}`
    : node.name;
  return match(matchText, filterText).length > 0;
};

const nodeOrChildrenMatch = (node, filterText) =>
  matcher(filterText, node) || // i match
  (node.children && // or i have decendents and one of them match
    node.children.length &&
    Boolean(
      node.children.find(childNode =>
        nodeOrChildrenMatch(childNode, filterText)
      )
    ));

const filterNode = (node, filterText) => {
  if (nodeOrChildrenMatch(node, filterText)) {
    const updatedNode = { ...node, expanded: true };
    if (updatedNode.children) {
      // eslint-disable-next-line no-use-before-define
      updatedNode.children = filterNodeArray(node.children, filterText);
    }
    return updatedNode;
  }
  return { ...node, expanded: false };
};

const filterNodeArray = (nodeArray, filterText) =>
  nodeArray.map(node => filterNode(node, filterText));

export default filterNodeArray;
