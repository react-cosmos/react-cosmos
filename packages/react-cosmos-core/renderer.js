module.exports = require('./dist/renderer/index.js');
