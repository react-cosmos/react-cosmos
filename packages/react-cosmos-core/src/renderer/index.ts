export * from './FixtureLoader/getDecoratedFixtureElement/index.js';
export * from './getFixtureList.js';
export * from './getSortedDecoratorsForFixturePath.js';
export * from './reactTypes.js';
export * from './registerPlaygroundShortcuts.js';
export * from './rendererConfig.js';
export * from './types.js';
