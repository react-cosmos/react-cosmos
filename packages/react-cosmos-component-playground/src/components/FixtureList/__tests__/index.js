import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import FixtureList from '../index';
import populatedFixture from '../__fixtures__/populated';
import populatedAndSelectedFixture from '../__fixtures__/populated-and-selected';

describe('Tree Fixtrue View', () => {
  test('should load Tree with propulated items ', () => {
    const list = renderer
      .create(<FixtureList {...populatedFixture.props} />)
      .toJSON();
    expect(list).toMatchSnapshot();
  });

  test('should load Tree with propulated items and selected path', () => {
    const list = renderer
      .create(<FixtureList {...populatedAndSelectedFixture.props} />)
      .toJSON();
    expect(list).toMatchSnapshot();
  });

  test.only('should have state set correctly', () => {
    const wrapper = shallow(
      <FixtureList {...populatedAndSelectedFixture.props} />
    );
    const instance = wrapper.instance();
    expect(instance.state.selectedPath).toEqual('ComponentA/bar');
    expect(instance.state.filteredFixtures).toEqual(
      populatedAndSelectedFixture.fixtures
    );
  });
});

// describe('Select', () => {
//   let wrapper;
//   let onUrlChange;

//   beforeEach(() => {
//     onUrlChange = jest.fn();
//     wrapper = shallow(<FixtureList {...populatedFixture.props} onUrlChange={onUrlChange} />);
//   });

//   test.only('should call select callback on click', () => {
//     const leaf = wrapper.find('.isLeaf');
//     leaf.simulate('click');
//     expect(onUrlChange.mock.calls.length).toEqual(1);
//     console.log(onUrlChange.mock.calls[0]);
//   });
// });

// describe('Search', () => {
//   let wrapper;
//   let searchInput;

//   beforeEach(() => {
//     wrapper = mount(<FixtureList {...populatedFixture.props} />);
//     searchInput = wrapper.find('.searchInput');
//   });

//   test('should only show matched component', () => {
//     searchInput.simulate('change', { target: { value: 'ux' } });
//     const instance = wrapper.instance();
//     expect(instance.state.searchText).toEqal('ux');
//     expect(instance.state.filteredFixtures).toEqal({
//       ComponentB: ['qux']
//     });
//   });
// });

describe('Search input keyboard shortcut', () => {
  let wrapper;
  let instance;

  // Triggering a window event is cumbersome...
  const triggerKeyEvent = (handler, keyCode) => {
    handler({ keyCode, preventDefault: jest.fn() });
  };

  beforeEach(() => {
    wrapper = mount(<FixtureList {...populatedFixture.props} />);
    instance = wrapper.instance();
  });

  describe('on `s` key', () => {
    let searchInput;

    beforeEach(() => {
      searchInput = wrapper.find('.searchInput');
      triggerKeyEvent(instance.onWindowKey, 83);
    });

    test('should focus input', () => {
      expect(searchInput.node).toBe(document.activeElement);
    });
  });

  describe('on ESC key', () => {
    let searchInput;

    beforeEach(() => {
      searchInput = wrapper.find('.searchInput');
      triggerKeyEvent(instance.onWindowKey, 83);
      searchInput.simulate('change', { target: { value: 'foo' } });
      triggerKeyEvent(instance.onWindowKey, 27);
    });

    test('should blur input', () => {
      expect(searchInput.node).not.toBe(document.activeElement);
    });

    test('should clear input', () => {
      expect(searchInput.prop('value')).toBe('');
    });
  });
});
