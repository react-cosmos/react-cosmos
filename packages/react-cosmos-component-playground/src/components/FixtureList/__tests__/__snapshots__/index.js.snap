// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree Fixtrue View should load Tree with propulated items  1`] = `
<div
  className="root"
>
  <div
    className="searchInputContainer"
  >
    <input
      className="searchInput"
      onChange={[Function]}
      placeholder="Search..."
      value=""
    />
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </div>
  <div
    className="nodeTree"
  >
    <div
      className="node"
    >
      <button
        className="nodeNameContainer isNode "
        data-path="ComponentA"
        onClick={[Function]}
        title="ComponentA"
      >
        <div
          className="nodeArrow"
        >
          <svg
            height="10px"
            viewBox="0 0 10 10"
            width="10px"
          >
            <path
              d="M 5,4.5 l0, 4.5, l-4,0z"
              fill="#eeeeee"
              stroke="#eeeeee"
            />
          </svg>
        </div>
        <div
          className="nodeIcon"
        >
          <svg
            height="15px"
            viewBox="0 0 20 20"
            width="18px"
          >
            <path
              d="M 0,2 l2,0 l 6,0 l2,2 l10,0 l0,15 l-20,0z"
              fill="#aaaaaa"
            />
          </svg>
        </div>
        <div
          className="nodeName"
        >
          ComponentA
        </div>
      </button>
      <div
        className="nodeChildren"
      >
        <div
          className="node"
        >
          <button
            className="nodeNameContainer isLeaf "
            data-path="ComponentA/foo"
            onClick={[Function]}
            title="ComponentA/foo"
          >
            <div
              className="nodeArrow"
            />
            <div
              className="nodeIcon"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgb(243, 204, 0)",
                    "color": "#333333",
                    "fontSize": "9px",
                    "height": "16px",
                    "left": "1px",
                    "position": "relative",
                    "width": "16px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "bottom": 0,
                      "position": "absolute",
                      "right": 1,
                    }
                  }
                >
                  JS
                </span>
              </div>
            </div>
            <div
              className="nodeName"
            >
              foo
            </div>
          </button>
        </div>
        <div
          className="node"
        >
          <button
            className="nodeNameContainer isLeaf "
            data-path="ComponentA/bar"
            onClick={[Function]}
            title="ComponentA/bar"
          >
            <div
              className="nodeArrow"
            />
            <div
              className="nodeIcon"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgb(243, 204, 0)",
                    "color": "#333333",
                    "fontSize": "9px",
                    "height": "16px",
                    "left": "1px",
                    "position": "relative",
                    "width": "16px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "bottom": 0,
                      "position": "absolute",
                      "right": 1,
                    }
                  }
                >
                  JS
                </span>
              </div>
            </div>
            <div
              className="nodeName"
            >
              bar
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="node"
    >
      <button
        className="nodeNameContainer isNode "
        data-path="ComponentB"
        onClick={[Function]}
        title="ComponentB"
      >
        <div
          className="nodeArrow"
        >
          <svg
            height="10px"
            viewBox="0 0 10 10"
            width="10px"
          >
            <path
              d="M 5,4.5 l0, 4.5, l-4,0z"
              fill="#eeeeee"
              stroke="#eeeeee"
            />
          </svg>
        </div>
        <div
          className="nodeIcon"
        >
          <svg
            height="15px"
            viewBox="0 0 20 20"
            width="18px"
          >
            <path
              d="M 0,2 l2,0 l 6,0 l2,2 l10,0 l0,15 l-20,0z"
              fill="#aaaaaa"
            />
          </svg>
        </div>
        <div
          className="nodeName"
        >
          ComponentB
        </div>
      </button>
      <div
        className="nodeChildren"
      >
        <div
          className="node"
        >
          <button
            className="nodeNameContainer isLeaf "
            data-path="ComponentB/baz"
            onClick={[Function]}
            title="ComponentB/baz"
          >
            <div
              className="nodeArrow"
            />
            <div
              className="nodeIcon"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgb(243, 204, 0)",
                    "color": "#333333",
                    "fontSize": "9px",
                    "height": "16px",
                    "left": "1px",
                    "position": "relative",
                    "width": "16px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "bottom": 0,
                      "position": "absolute",
                      "right": 1,
                    }
                  }
                >
                  JS
                </span>
              </div>
            </div>
            <div
              className="nodeName"
            >
              baz
            </div>
          </button>
        </div>
        <div
          className="node"
        >
          <button
            className="nodeNameContainer isLeaf "
            data-path="ComponentB/qux"
            onClick={[Function]}
            title="ComponentB/qux"
          >
            <div
              className="nodeArrow"
            />
            <div
              className="nodeIcon"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgb(243, 204, 0)",
                    "color": "#333333",
                    "fontSize": "9px",
                    "height": "16px",
                    "left": "1px",
                    "position": "relative",
                    "width": "16px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "bottom": 0,
                      "position": "absolute",
                      "right": 1,
                    }
                  }
                >
                  JS
                </span>
              </div>
            </div>
            <div
              className="nodeName"
            >
              qux
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tree Fixtrue View should load Tree with propulated items and selected path 1`] = `
<div
  className="root"
>
  <div
    className="searchInputContainer"
  >
    <input
      className="searchInput"
      onChange={[Function]}
      placeholder="Search..."
      value=""
    />
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </div>
  <div
    className="nodeTree"
  >
    <div
      className="node"
    >
      <button
        className="nodeNameContainer isNode "
        data-path="ComponentA"
        onClick={[Function]}
        title="ComponentA"
      >
        <div
          className="nodeArrow"
        >
          <svg
            height="10px"
            viewBox="0 0 10 10"
            width="10px"
          >
            <path
              d="M 5,4.5 l0, 4.5, l-4,0z"
              fill="#eeeeee"
              stroke="#eeeeee"
            />
          </svg>
        </div>
        <div
          className="nodeIcon"
        >
          <svg
            height="15px"
            viewBox="0 0 20 20"
            width="18px"
          >
            <path
              d="M 0,2 l2,0 l 6,0 l2,2 l10,0 l0,15 l-20,0z"
              fill="#aaaaaa"
            />
          </svg>
        </div>
        <div
          className="nodeName"
        >
          ComponentA
        </div>
      </button>
      <div
        className="nodeChildren"
      >
        <div
          className="node"
        >
          <button
            className="nodeNameContainer isLeaf "
            data-path="ComponentA/foo"
            onClick={[Function]}
            title="ComponentA/foo"
          >
            <div
              className="nodeArrow"
            />
            <div
              className="nodeIcon"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgb(243, 204, 0)",
                    "color": "#333333",
                    "fontSize": "9px",
                    "height": "16px",
                    "left": "1px",
                    "position": "relative",
                    "width": "16px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "bottom": 0,
                      "position": "absolute",
                      "right": 1,
                    }
                  }
                >
                  JS
                </span>
              </div>
            </div>
            <div
              className="nodeName"
            >
              foo
            </div>
          </button>
        </div>
        <div
          className="node"
        >
          <button
            className="nodeNameContainer isLeaf "
            data-path="ComponentA/bar"
            onClick={[Function]}
            title="ComponentA/bar"
          >
            <div
              className="nodeArrow"
            />
            <div
              className="nodeIcon"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgb(243, 204, 0)",
                    "color": "#333333",
                    "fontSize": "9px",
                    "height": "16px",
                    "left": "1px",
                    "position": "relative",
                    "width": "16px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "bottom": 0,
                      "position": "absolute",
                      "right": 1,
                    }
                  }
                >
                  JS
                </span>
              </div>
            </div>
            <div
              className="nodeName"
            >
              bar
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="node"
    >
      <button
        className="nodeNameContainer isNode "
        data-path="ComponentB"
        onClick={[Function]}
        title="ComponentB"
      >
        <div
          className="nodeArrow"
        >
          <svg
            height="10px"
            viewBox="0 0 10 10"
            width="10px"
          >
            <path
              d="M 5,4.5 l0, 4.5, l-4,0z"
              fill="#eeeeee"
              stroke="#eeeeee"
            />
          </svg>
        </div>
        <div
          className="nodeIcon"
        >
          <svg
            height="15px"
            viewBox="0 0 20 20"
            width="18px"
          >
            <path
              d="M 0,2 l2,0 l 6,0 l2,2 l10,0 l0,15 l-20,0z"
              fill="#aaaaaa"
            />
          </svg>
        </div>
        <div
          className="nodeName"
        >
          ComponentB
        </div>
      </button>
      <div
        className="nodeChildren"
      >
        <div
          className="node"
        >
          <button
            className="nodeNameContainer isLeaf "
            data-path="ComponentB/baz"
            onClick={[Function]}
            title="ComponentB/baz"
          >
            <div
              className="nodeArrow"
            />
            <div
              className="nodeIcon"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgb(243, 204, 0)",
                    "color": "#333333",
                    "fontSize": "9px",
                    "height": "16px",
                    "left": "1px",
                    "position": "relative",
                    "width": "16px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "bottom": 0,
                      "position": "absolute",
                      "right": 1,
                    }
                  }
                >
                  JS
                </span>
              </div>
            </div>
            <div
              className="nodeName"
            >
              baz
            </div>
          </button>
        </div>
        <div
          className="node"
        >
          <button
            className="nodeNameContainer isLeaf "
            data-path="ComponentB/qux"
            onClick={[Function]}
            title="ComponentB/qux"
          >
            <div
              className="nodeArrow"
            />
            <div
              className="nodeIcon"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgb(243, 204, 0)",
                    "color": "#333333",
                    "fontSize": "9px",
                    "height": "16px",
                    "left": "1px",
                    "position": "relative",
                    "width": "16px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "bottom": 0,
                      "position": "absolute",
                      "right": 1,
                    }
                  }
                >
                  JS
                </span>
              </div>
            </div>
            <div
              className="nodeName"
            >
              qux
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
